### PR TITLE
SPINEDEM-3894- search for a patient regression scenarios

### PR DESCRIPTION
**Pull request by agelledi**
## Summary
* Routine Change
* Adds additional search for a patient regression scenarios

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.

### CODE REVIEW

File: karate-tests/src/test/java/patients/healthcareWorker/searchForAPatient/searchParams.feature [hunk 1]
```diff
@@ -148,4 +148,182 @@ Scenario: Too many matches
...
+  * method get
+  * status 200
+  * match response.total == 0
+  # include history flag
+  * path 'Patient'
+  * params { family: '#(familyValue)', gender: '#(genderValue)', birthdate: '#(birthDateValue)', email: '#(emailValue)', _history: true }
+  * method get
+  * status 200
+  * def telecomValueList = karate.jsonPath(response, "$.entry[*].resource.telecom[*].value") 
+  * match each telecomValueList != emailValue
```
Comment (richardseacome1-nhs) [inline karate-tests/src/test/java/patients/healthcareWorker/searchForAPatient/searchParams.feature:246]: probably missing something here, but why are we checking that the email address is not in the telecoms list for historic

Comment (richardseacome1-nhs) [reply karate-tests/src/test/java/patients/healthcareWorker/searchForAPatient/searchParams.feature:246]: might also be good to inc. match response.total == 1

File: karate-tests/src/test/java/patients/healthcareWorker/searchForAPatient/searchParams.feature [hunk 2]
```diff
@@ -148,4 +148,182 @@ Scenario: Too many matches
...
+Scenario: Simple search with other given name - Single match
+  * def birthDateValue = "eq2000-05-05"
+  * def genderValue =  "male"
+  * def familyName = "Smith"
+  * def givenNames = ["Sam","Bob" ]
+  * path 'Patient'
+  * params { family: '#(familyName)', gender: '#(genderValue)', birthdate: '#(birthDateValue)', given: '#(givenNames)' }
+  * method get
+  * status 200
+  * assert response.total >= 1
```
[the remark this answers is not in the document]
Comment (agelledi) [reply karate-tests/src/test/java/patients/healthcareWorker/searchForAPatient/searchParams.feature:173]: Updated the assertion

File: karate-tests/src/test/java/patients/patientAccess/updatePatient.feature [hunk 3]
```diff
@@ -1,4 +1,3 @@
-@contactTest
```
Comment (richardseacome1-nhs) [inline karate-tests/src/test/java/patients/patientAccess/updatePatient.feature:1]: just curious, why is this deleted please?

File: karate-tests/src/test/java/patients/healthcareWorker/searchForAPatient/searchParams.feature (hunk 1 above)
Comment (agelledi) [reply karate-tests/src/test/java/patients/healthcareWorker/searchForAPatient/searchParams.feature:246]: Sorry about the confusion, this scenario uses the historical email address(previous email) along with history flag, I will update the scenario validations, thank you for spotting this

File: karate-tests/src/test/java/mocks/sandbox/get-patient-search.js [hunk 4]
```diff
@@ -125,21 +135,28 @@ if (request.pathMatches('/Patient') && request.get) {
...
   const email = request.param('email')
   const maxResults = request.param('_max-results')
+  const historyMatch = request.param('_history')
+  const gp = request.param('general-practitioner')
+  const deathDate = request.param('death-date')
 
   if (validateHeaders(request) && validateQueryParams(request)) {
     if (fuzzyMatch) {
-      if (family === 'Garton' && given[0] === 'Bill' && birthDate[0] === '1946-06-23') {
+      if (family === 'Garton' && (given[0]) === 'Bill' && birthDate[0] === '1946-06-23') {
```
Comment (richardseacome1-nhs) [inline karate-tests/src/test/java/mocks/sandbox/get-patient-search.js:152]: this if/else might be an easier read as a switch statement
https://www.w3schools.com/js/js_switch.asp

but I'm not going to mandate it, see if you think it's feasible

**Review by richardseacome1-nhs — COMMENTED** [review]
Few comments to address, generally good.

File: karate-tests/src/test/java/patients/patientAccess/updatePatient.feature (hunk 3 above)
Comment (agelledi) [reply karate-tests/src/test/java/patients/patientAccess/updatePatient.feature:1]: This annotation was added  for local debugging purpose in the previous pr.  It was unintentionally left in the code and is no longer needed, so I am removing it in this PR.

File: karate-tests/src/test/java/mocks/sandbox/get-patient-search.js (hunk 4 above)
Comment (agelledi) [reply karate-tests/src/test/java/mocks/sandbox/get-patient-search.js:152]: Agree, I am planning to create a separate task to make some improvements on mocks, I will try to implement suggested change as part of that ticket

**Review by richardseacome1-nhs — APPROVED** [review]
